src/login/pam_systemd.c

```diff
@@ -1317,6 +1320,10 @@ static int register_session(
          * somewhere else (for example PAM module parameters). Let's now update the environment variables, so that this
          * data is inherited into the session processes, and programs can rely on them to be initialized. */
 
+        _cleanup_free_ char *real_type = strdup(c->type); /* make copy because this might point to env block, which we are going to update shortly */
+        if (!real_type)
+                return pam_log_oom(handle);
+
```

Comment:
This should be done in a separate commit?? I cannot find anything related to OSC thing for this change.

Reply:
added a comment about this to the commit msg.

basically, we want to condition the OSC sequence on the session type we settled on. But the session type quite likely comes via a PAM environment variable to us and we just keep the pointer to it around. But we also change the PAM environment variable in the meantime invalidating the pointer. Hence we make a copy here now, simply because we still want to dereference the pointer still.